### PR TITLE
[7.x] Split testing component

### PR DESCRIPTION
**Pull request by crynobone**
So we can reused it on testing-browser-kit, lumen-framework etc.

---

Signed-off-by: Mior Muhammad Zaki <crynobone@gmail.com>

### CODE REVIEW

File: src/Illuminate/Testing/composer.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "phpunit/phpunit": "^8.4|^9.0"
+    },
+    "autoload": {
+        "psr-4": {
+            "Illuminate\\Testing\\": ""
+        }
+    },
+    "extra": {
+        "branch-alias": {
+            "dev-master": "7.0-dev"
```
Comment (driesvints) [inline src/Illuminate/Testing/composer.json]: This needs to be 
```suggestion
            "dev-master": "7.x-dev"
```

**Review by driesvints — CHANGES_REQUESTED** [review]
I like this idea 👍

File: src/Illuminate/Testing/composer.json (hunk 1 above)
Comment (crynobone) [reply src/Illuminate/Testing/composer.json]: I copy paste the value from `translation` composer.json

https://github.com/laravel/framework/blob/master/src/Illuminate/Translation/composer.json#L28-L31

Comment (driesvints) [reply src/Illuminate/Testing/composer.json]: Thanks for making me aware. I've fixed these on master 👍

**Review by driesvints — APPROVED** [review]

**Comment by crynobone** [comment]
Rebase PR, just realised I'm creating a duplicate instead of moving the classes.

File: src/Illuminate/Testing/composer.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    "authors": [
+        {
+            "name": "Taylor Otwell",
+            "email": "taylor@laravel.com"
+        }
+    ],
+    "require": {
+        "php": "^7.2.5",
+        "mockery/mockery": "^1.3.1",
```
Comment (GrahamCampbell) [inline src/Illuminate/Testing/composer.json]: Needs to also depend on `illuminate/console`, or at least suggest it, due to https://github.com/crynobone/framework/blob/testing-component/src/Illuminate/Testing/PendingCommand.php#L162-L164.

Also, need to depend on the illuminate/contracts repo?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Testing/composer.json (hunk 2 above)
Comment (GrahamCampbell) [reply src/Illuminate/Testing/composer.json]: Also need illuminate/support for all the TestResponse, and should "suggest" illuminate/database.

Comment (crynobone) [reply src/Illuminate/Testing/composer.json]: I would agree we need to require `illuminate/contracts` and `illuminate/support`.

But would be based if `illuminate/console` (for `PendingCommand`), `illuminate/http` (for `TestResponse`) and `illuminate/database` (for database assertions) should just be suggest.

Comment (crynobone) [reply src/Illuminate/Testing/composer.json]: For example you might want to use this package to test `illuminate/database` running outside of Laravel or Lumen (using capsule).

I not sure how useful PendingCommand or TestResponse on similar usage right now.

Comment (GrahamCampbell) [reply src/Illuminate/Testing/composer.json]: 👍

File: src/Illuminate/Testing/composer.json [hunk 3]
```diff
@@ -15,6 +15,8 @@
     ],
     "require": {
         "php": "^7.2.5",
+        "illuminate/contracts": "^6.0",
```
Comment (GrahamCampbell) [inline src/Illuminate/Testing/composer.json]: Should be `^7.0`. ;)

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Maybe it's worth moving some of the other stuff. Like `WithoutFaker`?

**Comment by crynobone** [comment]
> Maybe it's worth moving some of the other stuff. Like WithoutFaker?

Not sure about that since that would requires everyone to update the classname import for every tests that manually import them.

**Pull request merged**